### ISSUE TITLE
Get complete list of opt-in diagnostics

### ISSUE THREAD

**Issue by matthewpaulthomas** [CanonicalLtd/desktop-design#57]
@matthewpaulthomas commented on [Wed Feb 07 2018](https://github.com/CanonicalLtd/snappy-design-squad/issues/359)

**Comment by matthewpaulthomas** [comment]
I’ve mailed Will to ask what diagnostics we will be collecting.

**Comment by 8none1** [comment]
This Trello card should have everything you need:

https://trello.com/c/QBufNYWO/174-add-a-send-data-to-help-improve-ubuntu-option-to-the-installer-and-to-settings

UI wise, we would like a mechanism in the installer to describe and allow opt out (We would like it on by default because we will then get crash reports which are very valuable to us to spot problems).

The privacy panel in the new (to us) GNOME Settings would need a corresponding mechanism.

**Comment by matthewpaulthomas** [comment]
@8none1 Thanks. Oddly I can subscribe to that card but can’t comment on it. So a small correction here:
> Additional: If the checkbox is on, then apport should be configured to "just send" and not nag the user. If it's off, dont send apport.

This is a common misunderstanding from those with an engineering POV. For me, the error-reporting dialog was a lucky opportunity to fix a separate problem in Ubuntu. [As I explained in the spec in 2012](https://wiki.ubuntu.com/ErrorTracker#Rationale), “The client interface for the error tracker also serves a purpose which is less important for developers, but _more_ important for end users: *explaining why something weird just happened*. In previous Ubuntu release versions, when most programs crashed there was no explanation of the cause.”

In other words, _even if we had no error reporting at all_, when an app crashed, it would still be better to have a dialog explaining why the app just disappeared — and making the app easy to relaunch — than not to have one. Ubuntu engineers often misunderstand this, because they understand crashes, and because they happen to know which bit of software is providing the dialog.

**Comment by 8none1** [comment]
@matthewpaulthomas you should have access now.
Please could you comment on there too?

**Comment by matthewpaulthomas** [comment]
Done.

**Comment by seb128** [comment]
> In other words, even if we had no error reporting at all, when an app crashed, it would still be better to have a dialog explaining why the app just disappeared — and making the app easy to relaunch — than not to have one. Ubuntu engineers often misunderstand this

Saying that those who argue against showing those prompts misunderstand the situation is not accurate, most of us do understand the benefits. The issue is that today most of those dialogs prompt for issues that have no (user visible) impact, often for errors that the system handled fine by retry/respawning the corresponding service. Some are displayed in the next session to report about problems that were hit on the previous closing/shutdown. Even when the dialog is about an application that closed hitting an error the timing and placement of the error doesn't really help users in an obvious way.

Yes, those are technical issues and solvable. Your design/suggestions are probably adequate, but those problems have not been addressed in 6 years, there is no reason to believe the issues are going to be resolved this cycle. Those who argue for removing those prompt don't do it because they misunderstand the purpose, just because they think the currently implementation is having enough problems that the result is detrimental to users rather than helping them.

**Comment by popey** [comment]
We are talking about these dialogs, yes?

![image](https://user-images.githubusercontent.com/1841272/36280514-678ba3e4-1292-11e8-8895-9fc9639af123.png)

![image](https://user-images.githubusercontent.com/1841272/36280533-7576f576-1292-11e8-89ce-b576368e3975.png)

We get comments *all* the time from users who have visceral hatred for the "something crashed" popup. They see it as a sign that Ubuntu doesn't any QA, that Ubuntu is poor quality implementation of a Linux distribution & that Ubuntu crashes more than it used to.

Users want to [disable](https://askubuntu.com/questions/93457/how-do-i-enable-or-disable-apport) it, make [videos](https://www.youtube.com/watch?v=iRSUEloyPsM) detailing out to disable it, write [blog](http://howtoubuntu.org/how-to-disable-stop-uninstall-apport-error-reporting-in-ubuntu) [posts](https://itsfoss.com/how-to-fix-system-program-problem-detected-ubuntu/) about how to disable it and generally encourage the world not to send crash dumps to us. 

We have over the years argued that the dialog is a useful representation of software or hardware that needs to be fixed, and users are empowered to tell us that. Most though frankly don't care. They want the dialog to go away and never come back and for us to fix the broken stuff which caused it to pop up.

If there was one thing I'd dearly love to see fixed in 18.04, it's that dialog to go away and the crashes just go to errors.ubuntu.com by default, opted in by the user.
